### PR TITLE
Add WoT-Con (OPC 10100-1) companion spec mock

### PR DESCRIPTION
**Pull request by karok2m**
Adds a mock-only, server-side implementation of the **OPC UA Web of Things Connectivity (WoT-Con) companion spec** ([OPC 10100-1 v1.02](https://reference.opcfoundation.org/specs/OPC-10100-1/v1.02/6.3)) under `src/CompanionSpecs/WotCon/`. Gives consumers (Commander, dashboards, E2E tests) a spec-compliant address space to drive against without standing up a real southbound binding.

Off by default — enable with `--wotcon`.

## What's in scope (§6.3.1 – §6.3.11)

- `WoTAssetConnectionManagement` entry point.
- `CreateAsset` / `DeleteAsset` with strict `Bad_BrowseNameDuplicated` and `Bad_NotFound`.
- Optional members materialized on the management object: `SupportedWoTBindings`, `DiscoverAssets`, `CreateAssetForEndpoint`, `ConnectionTest`, `Configuration` / `License`.
- Per-asset `WoTAssetFileType` instance with the full OPC 10000-5 `FileType` layout plus the WoT-Con `CloseAndUpdate` extension.
- `CloseAndUpdate` parses the uploaded Thing Description and materializes its **properties as `BaseDataVariableState`s** and **actions as `MethodState`s** under the asset, linked via `HasWoTComponent` (§6.3.11).
- JSON Schema primitives map to OPC UA built-in types per §6.3.8 Table 14; `unit` → `EngineeringUnits`; `readOnly` / `writeOnly` flip `AccessLevel`; `base` URI surfaces as the `AssetEndpoint` Property and feeds `DiscoverAssets`.
- Re-upload drops the previous generation via `DeleteNode` before re-materializing.

## What's deferred

Mock simulation engine (live value drift), structured types (nested object schemas fall back to JSON-as-String), and authorization stubs — all gated on a real Commander consumer. The README covers these explicitly so nothing is silently absent.

## Out of scope

Real Modbus / HTTP / OPC DA bindings, WoT Event affordances (v1.02 defines no mapping), TD authoring / TDD federation, cross-asset hierarchies.

## Tests

- 38 new tests under `tests/CompanionSpecs/WotCon/`, split into per-theme partials (`CreateAsset`, `DeleteAsset`, `CloseAndUpdate`, `WoTFile`, `PropertyMaterialization`, `ActionMaterialization`, `HasWoTComponent`, `DiscoverAssets`, `OptionalMembers`, `Discovery`).
- Coverage includes a structural address-space audit of the per-asset `WoTAssetFileType` (the kind of test that would have caught the historical missing-`HasComponent` regression — handler-call tests pass on NodeId, but the address space was empty).
- E2E: the [`microsoft-opc-plc`](https://msazure.visualstudio.com/One/_git/iotcat?path=/aio-connectors-tooling/distrib/helm/microsoft-opc-plc) helm chart passes `--wotcon` and the `opcplc-opc-plc-wot-000000` deployment exposes `CreateAsset` end-to-end.

See [`src/CompanionSpecs/WotCon/README.md`](src/CompanionSpecs/WotCon/README.md) for the spec mapping table, architecture diagram, design notes, and links.

### CODE REVIEW

File: src/CompanionSpecs/WotCon/WotConNodeManager.cs [hunk 1]
```diff
@@ -0,0 +1,1438 @@
...
+            return ServiceResult.Good;
+        }
+        catch (Exception ex)
+        {
+            _logger?.LogError(ex, "[WotCon] {Asset}.OnPerAssetFileOpen failed", asset.Name);
+            return new ServiceResult(StatusCodes.BadInternalError, ex.Message);
+        }
+    }
+
+    private ServiceResult OnPerAssetFileWrite(WotAsset asset, FileState fileNode, IList<object> inputArguments)
```
Comment (dacolgit) [inline src/CompanionSpecs/WotCon/WotConNodeManager.cs:937]: The `fileNode.MaxByteStringLength` is never checked.
A client can append unbounded data and OOM the simulator while the node claims a limit

File: src/CompanionSpecs/WotCon/WotConNodeManager.cs [hunk 2]
```diff
@@ -0,0 +1,1438 @@
...
+    // (Variables for Properties, Methods for Actions). Generic HasComponent stays in
+    // use for non-affordance plumbing such as the per-asset WoTFile.
+    private const uint HasWoTComponentReferenceTypeId = 142;
+
+    // Strict UTF-8 decoder: throws DecoderFallbackException on malformed byte sequences
+    // instead of silently substituting U+FFFD. Used to validate uploaded TD payloads.
+    private static readonly UTF8Encoding StrictUtf8 = new(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+
+    private readonly ILogger _logger;
+    private readonly Dictionary<string, WotAsset> _assets;
```
Comment (dacolgit) [inline src/CompanionSpecs/WotCon/WotConNodeManager.cs]: I think it's safer to guard every access to _assets and _filesByNodeId with a lock. Concurrent requests from clients can result in OnCreateAsset/OnDeleteAsset running in parallel on different threads, and since both are plain Dictionarys this can lead to corrupted state

Comment (karok2m) [reply src/CompanionSpecs/WotCon/WotConNodeManager.cs]: implemented.

File: src/CompanionSpecs/WotCon/WotConNodeManager.cs (hunk 1 above)
Comment (karok2m) [reply src/CompanionSpecs/WotCon/WotConNodeManager.cs:937]: Implemented.

**Comment by luiscantero** [comment]
LGTM, but you need to increase version.json to be able to release

**Review by luiscantero — APPROVED** [review]

**Review by dacolgit — APPROVED** [review]
